### ISSUE TITLE
Text copied from last line of file gets pasted to the wrong place

### ISSUE THREAD

**Issue by jdw1996** [atom/atom#14083]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When I copy a line without selecting any part of it, whether using Ctrl-c or right clicking and choosing "Copy", and then try to paste the contents of the line, they get added to the beginning of the line instead of being put where my cursor is.

### Steps to Reproduce

1. Open a file in Atom. It doesn't matter if there are other lines in this file or not.
2. Move your cursor to the very last line of the file. Type some text there. Press Ctrl-c.
3. Place your cursor on any line in the file, at any place in the line, and press Ctrl-v.

**Expected behavior:** [What you expect to happen]

I would expect the copied text to be pasted where the cursor is, without any newlines being pasted.

**Actual behavior:** [What actually happens]

Instead, the copied text is added to the beginning of the line and, if the cursor is not on the last line of the file, it moves down a line.

Here's a simple example, where I copy `a`, type `b`, then paste repeatedly:

![copy-paste-bug](https://cloud.githubusercontent.com/assets/17225098/24438786/e53b6fd8-1417-11e7-9fd5-62bacd7a0fec.gif)

**Reproduces how often:** [What percentage of the time does it reproduce?]

This happens every time.

### Versions

```
$ atom --version
Atom    : 1.15.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0
$ apm --version
apm  1.15.3
npm  3.10.5
node 4.4.5 x64
python 2.7.12
git 2.7.4
```

I'm using Ubuntu 16.04, but I've seen the same issue in Windows 7 as well.

**Comment by rsese** [comment]
> 1. Open a file in Atom. It doesn't matter if there are other lines in this file or not.
> 2. Move your cursor to the very last line of the file. Type some text there. Press Ctrl-c.

Just to confirm, at this point you don't select any text you just use <kbd>Ctrl+c</kbd>?

**Comment by rsese** [comment]
Maybe related-ish /cc https://github.com/atom/atom/issues/11542?

**Comment by jdw1996** [comment]
@rsese That's correct, this behaviour comes up when I try copying without selecting anything.

Upon playing around with this a bit more, I realized that there is similar unexpected behaviour when I do this with lines other than the last one, too:

![copy-paste-bug](https://cloud.githubusercontent.com/assets/17225098/24486352/c08743d6-14d7-11e7-97b6-0c0dfcffe5b7.gif)

I get the same effect when cutting with Ctrl-x instead of copying.

I tested out the same sequences of commands in Gedit and Emacs, and both of those copy the contents of the line without any newline, and then paste at the cursor.

Sorry if any of this is still unclear; I'm happy to add more explanation if need be.

**Comment by rsese** [comment]
Thanks for confirming, I see the same behavior you're describing - I did note that in the first case where you're at the last line of a file (or the only line in a single line file) there's no newline but in the second case there is with the situation you have (I had happened to have `Show invisibles` enabled).  

So copying the line with no selection is working fine, the question you have is about the paste behavior?  You mentioned the behavior of Gedit and Emacs so I tried things out in Sublime just to try another editor.  In case 1, the behavior is different than Atom:

![sublime-copy-line-1](https://cloud.githubusercontent.com/assets/734194/24534165/cbb34236-1591-11e7-9407-c8d334030659.gif)

It seems to behave as if there is a newline in the last line unlike Atom where it looks like there's not but the paste isn't at the cursor.

In the second case you described, the behavior is the same as Atom:

![sublime-copy-line-2](https://cloud.githubusercontent.com/assets/734194/24534077/583ae926-1591-11e7-836c-29905a01dcb3.gif)

In this case, the paste isn't at the cursor either.

So I guess the question is if this behavior is buggy or works as expected - can you confirm @lee-dohm?

**Comment by jdw1996** [comment]
That is interesting about Sublime. I would have checked it, but I don't have it installed.

The Gedit/Emacs behaviour seems more intuitive to me, but that's likely because those are the tools I'm used to using. I think it would be nice for users to be able to choose which behaviour they want, but programming that may be more trouble than it's worth, and I definitely understand if tuning this relatively rare case is not a priority.

**Comment by lee-dohm** [comment]
@rsese I would say the text pasted should be at the point of the cursor, no matter what is in the clipboard and where the cursor is located. Behavior other than that is a bug.

**Comment by rsese** [comment]
Thanks for the 👓 @lee-dohm :v:  I'll go ahead and label as a bug then since the paste isn't happening at the point of the cursor in both cases.

With that @jdw1996, seems like the correct behavior and the behavior you expect is that in case 1:

1. on the last line of a file, type <kbd>a</kbd>
1. cut the line with <kbd>Ctrl+x</kbd>
1. type <kbd>b</kbd>
1. place the cursor after `b`
1. paste with <kbd>Ctrl-v</kbd>

The result would be:

```
ba
```

Further <kbd>Ctrl-v</kbd> would keep on appending `a` i.e.:

```
baaa
```

In case 2:

1. type <kbd>a</kbd>
1. add a newline with <kbd>Enter</kbd>
1. type <kbd>b</kbd>
1. go to the line with `a` and copy the line with <kbd>Ctrl+c</kbd>
1. go to the line with `b` and place the cursor after the `b`
1. paste with <kbd>Ctrl-v</kbd>

The result would be:

```
a
ba

```

Further <kbd>Ctrl-v</kbd> would keep on appending `a` followed by a newline i.e.:

```
a
ba
a
a
a
```

Let me know if any of this is incorrect :v:

**Comment by jdw1996** [comment]
@rsese You've described my expectations exactly. Thanks!
